### PR TITLE
Backport multiple paginators

### PR DESCRIPTION
**Pull request by chav170**
Backporting multiple paginators feature from cakephp 3.3.

Refs #8488

### CODE REVIEW

File: lib/Cake/Controller/Component/PaginatorComponent.php [hunk 1]
```diff
@@ -225,7 +241,8 @@ public function paginate($object = null, $scope = array(), $whitelist = array())
 			'order' => $order,
 			'limit' => $limit,
 			'options' => Hash::diff($options, $defaults),
-			'paramType' => $options['paramType']
+			'paramType' => $options['paramType'],
+			'queryScope' => Hash::get($options, 'queryScope'),
```
Comment (andrej-griniuk) [inline lib/Cake/Controller/Component/PaginatorComponent.php]: Please add `queryScope` to `getDefaults()`. Then you can use it here via `$options['queryScope']` instead, just for consistency.

File: lib/Cake/Controller/Component/PaginatorComponent.php [hunk 2]
```diff
@@ -317,6 +334,10 @@ public function mergeOptions($alias) {
 				$request = $this->Controller->request->query;
 				break;
 		}
+		$queryScope = Hash::get($defaults, 'queryScope');
```
Comment (andrej-griniuk) [inline lib/Cake/Controller/Component/PaginatorComponent.php]: `$defaults['queryScope']` as well.

File: lib/Cake/Test/Case/View/Helper/PaginatorHelperTest.php [hunk 3]
```diff
@@ -2829,6 +2928,12 @@ public function testCurrent() {
 	public function testNoDefaultModel() {
 		$this->Paginator->request = new CakeRequest(null, false);
 		$this->assertNull($this->Paginator->defaultModel());
+
+		$this->Paginator->defaultModel('Article');
```
Comment (andrej-griniuk) [inline lib/Cake/Test/Case/View/Helper/PaginatorHelperTest.php]: I would move this to a separate test function, cause now `testNoDefaultModel()` name doesn't match what's being tested.

**Comment by chav170** [comment]
Created PR to the core, closing this.

https://github.com/cakephp/cakephp/pull/9705

**Pull request closed without merging**